00X-ABI-Stable-Module-API.md

```diff
@@ -0,0 +1,377 @@
+NODE_EXTERN napi_value napi_create_symbol(napi_env e, const char* s = NULL);
+NODE_EXTERN napi_value napi_create_function(napi_env e, napi_callback cbinfo);
+NODE_EXTERN napi_value napi_create_error(napi_env e, napi_value msg);
+NODE_EXTERN napi_value napi_create_type_error(napi_env e, napi_value msg);
+
+
+// Methods to get the the native napi_value from Primitive type
+NODE_EXTERN napi_valuetype napi_get_type_of_value(napi_env e, napi_value v);
+NODE_EXTERN double napi_get_number_from_value(napi_env e, napi_value v);
+NODE_EXTERN int napi_get_string_from_value(napi_env e, napi_value v,
+                                           char* buf, const int buf_size);
+NODE_EXTERN int32_t napi_get_value_int32(napi_env e, napi_value v);
+NODE_EXTERN uint32_t napi_get_value_uint32(napi_env e, napi_value v);
+NODE_EXTERN int64_t napi_get_value_int64(napi_env e, napi_value v);
+NODE_EXTERN bool napi_get_value_bool(napi_env e, napi_value v);
+
+NODE_EXTERN int napi_get_string_length(napi_env e, napi_value v);
+NODE_EXTERN int napi_get_string_utf8_length(napi_env e, napi_value v);
+NODE_EXTERN int napi_get_string_utf8(napi_env e, napi_value v,
+                                     char* buf, int bufsize);
+
+
+// Methods to coerce values
+// These APIs may execute user script
+NODE_EXTERN napi_value napi_coerce_to_object(napi_env e, napi_value v);
+NODE_EXTERN napi_value napi_coerce_to_string(napi_env e, napi_value v);
+
+
+// Methods to work with Objects
+NODE_EXTERN napi_value napi_get_prototype(napi_env e, napi_value object);
+NODE_EXTERN napi_propertyname napi_property_name(napi_env e,
+                                                 const char* utf8name);
+NODE_EXTERN napi_value napi_get_propertynames(napi_env e, napi_value object);
+NODE_EXTERN void napi_set_property(napi_env e, napi_value object,
+                                   napi_propertyname name, napi_value v);
+NODE_EXTERN bool napi_has_property(napi_env e, napi_value object,
+                                   napi_propertyname name);
+NODE_EXTERN napi_value napi_get_property(napi_env e, napi_value object,
+                                         napi_propertyname name);
+NODE_EXTERN void napi_set_element(napi_env e, napi_value object,
+                                  uint32_t i, napi_value v);
+NODE_EXTERN bool napi_has_element(napi_env e, napi_value object, uint32_t i);
+NODE_EXTERN napi_value napi_get_element(napi_env e,
+                                        napi_value object, uint32_t i);
+
+
+// Methods to work with Arrays
+NODE_EXTERN bool napi_is_array(napi_env e, napi_value v);
+NODE_EXTERN uint32_t napi_get_array_length(napi_env e, napi_value v);
+
+// Methods to compare values
+NODE_EXTERN bool napi_strict_equals(napi_env e, napi_value lhs, napi_value rhs);
+
+
+// Methods to work with Functions
+NODE_EXTERN void napi_set_function_name(napi_env e, napi_value func,
+                                        napi_propertyname napi_value);
```

Comment:
Created to https://github.com/nodejs/abi-stable-node/issues/26 to investigate/figure out  if we need a different API here.